OIIInspector/image_manager.py

```diff
@@ -0,0 +1,187 @@
+import logging
+from retry import retry
+import subprocess
+import time
+import shlex
+import OIIInspector.exceptions as exceptions
+from OIIInspector.utils import run_cmd
+
+log = logging.getLogger(__name__)
+
+
+class ImageManager:
+    """
+    Class used as context manager for image operations.
+    """
+    _grpc_start_port = 50051
+    _grpc_max_port_tries = 100
+    _grpc_init_wait_time = 3
+    _base_container_name = "OIIInspector_running_image"
+
+    _container_platform = None
+
+    def __init__(self, image_address):
+        """
+        Initialization of the ImageManager
+
+        :param image_address:
+        :type image_address:
+        """
+        self._image_address = None
+        self._port = self._grpc_start_port
+        self._rpc_proc = None
+        self._image_address = image_address
+        self._image_is_running = False
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_tb):
+        self.close_image_manager()
+
+    def start_image(self):
+        """
+        Method that starts desired container at _grpc_start_port
+        """
+        # pull needs to be done separately, because command run (which is capable of pull), would be terminated by timer
+
+        if self._container_platform is None:
+            self._get_container_platform()
+        self._pull_image()
+        self._port, self._rpc_proc = self._serve_index_registry()
+        self._image_is_running = True
+
+    def get_local_address_of_image(self) -> str:
+        """
+        Method for getting local address of the running image
+
+        :return: Returns local address where image is running in format localhost:PORT
+        :rtype: str
+        """
+        if self._image_is_running is False:
+            raise (exceptions.OIIInspectorError("Image is not running"))
+        return f"localhost:{self._port}"
+
+    def close_image_manager(self):
+        """
+        Close and remove running image
+        """
+        if self._image_is_running:
+            self._stop_container()
+            self._rpc_proc.kill()
```

Comment:
This is the "docker run" process, right? Shouldn't we kill it before stopping the container with another command?